keras/src/layers/layer.py

```diff
@@ -679,14 +690,24 @@ def set_weights(self, weights):
 
     @property
     def dtype_policy(self):
-        return self._dtype_policy
+        if isinstance(self._dtype_policy, DTypePolicyMap) and self.path:
```

Comment:
Doesn't this break things like `MultiHeadAttention`? In the changes I was playing with, we only resolved `path` in the `compute_dtype`, `variable_dtype`, and `quatization_mode` properties. That was so that composite layer could just pass down the dtype policy to children. Like this

https://github.com/keras-team/keras/blob/2f846b0cfeafbe9004e727c571f0e496057201b1/keras/src/layers/attention/multi_head_attention.py#L295

If we want `dtype=self.dtype_policy` to be the way to propagate dtype, I think we need to flip back to how this was--only resolve path in `compute_dtype`, `variable_dtype`, and `quatization_mode` getters.

Reply:
Thanks! I wasn't aware of that.
I have added a test in MHA to prevent future breakage.